vnext/Shared/JSI/NapiJsiRuntime_detail.h

```diff
@@ -0,0 +1,505 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+// JSI
+#include <js_native_ext_api.h>
+#include <jsi/jsi.h>
+
+// Standard Library
+#include <string_view>
+
+#ifdef __cpp_lib_span
+#include std::span;
+#endif // __cpp_lib_span
+
+namespace Microsoft::JSI {
+
+namespace {
```

Comment:
Anonymous namespaces must be never used in the header files. It produces a not-optimal code.